### PR TITLE
♻️ Added query params parameter to HTTP get requests

### PR DESCRIPTION
**Pull request by DarviL82**
### Changes

-   `adds`: The ability to specify a `params` keyword argument in the HTTP get request, which will add the supplied dictionary to the query url, instead of doing it manually on each endpoint that requires this.
-   `fixed`: `Guild.search_guild_members` not working due to an incorrect variable in the request url.
-   `improvements`: All methods using query parameters were refactored for using this.

 ### Check off the following

-   [X] I have tested my changes with the current requirements
-   [X] My Code follows the pep8 code style.

### CODE REVIEW

**Review by Lunarmagpie — APPROVED** [review]
lgtm. Im pretty sure moving `remove_none` to the http class should be fine.

File: pincer/core/http.py [hunk 1]
```diff
@@ -133,6 +135,12 @@ async def __send(
...
 
+        headers:
+            The request headers.
+
+        params:
+            The query parameters to add to the request.
+
         __ttl:
             Private param used for recursively setting the retry amount.
             (Eg set to 1 for 1 max retry)
```
Comment (Endercheif) [inline pincer/core/http.py:138]: I know this isn't fully your fault but can you add types to the docstrings

File: pincer/core/http.py [hunk 2]
```diff
@@ -301,7 +310,11 @@ async def delete(
             headers=headers
         )
 
-    async def get(self, route: str) -> Optional[Dict]:
+    async def get(
+        self,
+        route: str,
+        params: Optional[dict] = None
```
Comment (Endercheif) [inline pincer/core/http.py:316]:
```suggestion
        params: Optional[Dict] = None
```

File: pincer/core/http.py [hunk 3]
```diff
@@ -310,13 +323,20 @@ async def get(self, route: str) -> Optional[Dict]:
         ----------
         route : :class:`str`
             The Discord REST endpoint to send a get request to.
+        params: Optional[:class:`dict`]
```
Comment (Endercheif) [inline pincer/core/http.py:326]:
```suggestion
        params: Optional[:class:`Dict`]
```

File: pincer/core/http.py [hunk 4]
```diff
@@ -310,13 +323,20 @@ async def get(self, route: str) -> Optional[Dict]:
...
+            The query parameters to add to the request.
+            |default| :data:`None`
 
         Returns
         -------
         Optional[:class:`dict`]
```
Comment (Endercheif) [inline pincer/core/http.py:316]: pls change to `Dict` instead of `Dict`

**Review by Endercheif — CHANGES_REQUESTED** [review]

**Review by trag1c — APPROVED** [review]

**Pull request merged**